qiskit/circuit/quantumcircuit.py

```diff
@@ -955,30 +958,41 @@ def compose(
                 )
             edge_map.update(zip(other.clbits, dest.cbit_argument_conversion(clbits)))
 
+        register_map = {}
```

Comment:
In my first read, I didn't realize that this was passed as a parameter to `map_register_to_dest`. Perhaps it'd improve readability if this were to be declared below this function so it is not shadowed by the parameter.

Reply:
I slightly modified the function to close over the variable instead, calling it `_map_register_cache` in 8b90551c6.  I guess I originally passed it as a parameter because that's just how I prefer to treat mutable state, but the flip side of that is that I usually much prefer to _return_ the mutable state as well, and that would make the call sites really ugly here.